hazelcast/src/main/java/com/hazelcast/internal/config/ConfigValidator.java

```diff
@@ -156,14 +178,15 @@ public static void checkMapEvictionConfig(MaxSizePolicy maxSizePolicy,
                                               String comparatorClassName,
                                               Object comparator) {
         checkEvictionConfig(evictionPolicy, comparatorClassName,
-                comparator, IMAP_SUPPORTED_EVICTION_POLICIES);
+                comparator, MAP_SUPPORTED_EVICTION_POLICIES);
         checkMapMaxSizePolicyConfig(maxSizePolicy);
     }
 
     private static void checkMapMaxSizePolicyConfig(MaxSizePolicy maxSizePolicy) {
-        if (!MAP_SUPPORTED_ALL_MAX_SIZE_POLICIES.contains(maxSizePolicy)) {
+        if (!MAP_SUPPORTED_ON_HEAP_MAX_SIZE_POLICIES.contains(maxSizePolicy)
+                && !MAP_SUPPORTED_NATIVE_MAX_SIZE_POLICIES.contains(maxSizePolicy)) {
             String msg = format("IMap eviction config doesn't support max size policy `%s`. "
-                    + "Please select a valid one: %s.", maxSizePolicy, MAP_SUPPORTED_ALL_MAX_SIZE_POLICIES);
+                    + "Please select a valid one: %s.", maxSizePolicy, MAP_SUPPORTED_ON_HEAP_MAX_SIZE_POLICIES);
```

Comment:
Should we not print out all of the policies here, since the check allows for `MAP_SUPPORTED_NATIVE_MAX_SIZE_POLICIES` too?

Reply:
good catch, fixed.